libs/queue/src/main.sw

```diff
@@ -0,0 +1,98 @@
+// The `Queue` type corresponds to the same data structure.
+library;
+
+use std::vec::Vec;
+
+pub struct Queue<T> {
+    vec: Vec<T>,
+}
+
+impl<T> Queue<T> {
+    /// Create a new `Queue`
+    pub fn new() -> Self {
+        Self {
+            vec: Vec::new(),
+        }
+    }
+
+    /// Checks a `Queue` for emptiness
+    pub fn is_empty(self) -> bool {
+        self.vec.is_empty()
+    }
+
+    /// Gets the number of elements in the `Queue`
+    pub fn len(self) -> u64 {
+        self.vec.len()
+    }
+
+    /// Enqueues a `Queue`
+    pub fn enqueue(ref mut self, item: T) {
+        self.vec.push(item);
+    }
+
+    /// Dequeues a `Queue`
+    pub fn dequeue(ref mut self) -> Option<T> {
+        if self.vec.is_empty() {
+            Option::None
+        } else {
+            Option::Some(self.vec.remove(0))
+        }
+    }
```

Comment:
```suggestion
    /// Dequeues a `Queue`
    pub fn dequeue(ref mut self) -> Option<T> {
        if self.vec.is_empty() {
            return Option::None;
        }
        
        return Option::Some(self.vec.remove(0));
    }
```

Usually, it's better to structure code such that unnecessary nesting / indentation can be removed, this is particularly important when there are multiple levels of nested code.

Reply:
Fixed as outlined, though IMHO returns that break the flow are worse that nested code, but that's proobably up to personal taste, at least here